src/effndof.jl

```
@@ -0,0 +1,33 @@
+"""
+    effndof(μ)
+
+Returns the effective number of degrees of freedom of variates of
+measure-like object `μ`.
+
+The effective NDOF my differ from the length of the variates. For example,
+the effective NDOF for a Dirichlet distribution with variates of length `n`
+is `n - 1`.
+
+Also see [`require_same_effndof`](@ref).
+"""
+function effndof end
+
+
+"""
+    MeasureBase.require_same_effndof(a, b)::Nothing
+
+Check if `a` and `b` have the same effective number of degrees of freedom
+according to [`MeasureBase.effndof`](@ref).
+"""
+function require_same_effndof end
+
+ChainRulesCore.rrule(::typeof(require_same_effndof), a, b) = nothing, _nogradient_pullback2
+
+function require_same_effndof(a, b)
```

Can you tell me more about how these `require_*` functions are used?

---

`require_same_effndof` is used in the `VarTransformation` constructor and the generic `vartransform` to fail early if things aren't compatible.

The `rrule` is there to remove the cost of that check from the pullback, of course.

---

Called `check_dof` now.